src/StockportWebapp/Views/stockportgov/Shared/GenericFeaturedItemList.cshtml

```diff
@@ -1,42 +1,45 @@
 ﻿
 @model StockportWebapp.Models.GenericFeaturedItemList
 
-<div class="primary-topics">
-    @if (Model.HideButton == true)
-    {
-        <div class="featured-topic-list generic-Featured-Items-margin-top matchbox-parent">
-            <partial name="GenericFeaturedItem" model='Model.Items' />
-        </div>
-    }
-    else
-    {
-        @for (var count = 0; count < 8; count += 4)
+    <div class="primary-topics">
+        @if (Model.HideButton)
         {
-            <div class="featured-topic-list generic-Featured-Items-margin-top matchbox-parent">
-                <partial name="GenericFeaturedItem" model='Model.Items.Skip(count).Take(4)' />
-            </div>
+            @for (var count = 0; count < Model.Items.Count(); count += 4)
+            {
+                <div class="featured-topic-list generic-Featured-Items-margin-top matchbox-parent">
+                    <partial name="GenericFeaturedItem" model='Model.Items.Skip(count).Take(4)' />
+                </div>
+             }
+        } 
+        else 
+        {
+            @for (var count = 0; count < 8; count += 4)
+            {
+                <div class="featured-topic-list generic-Featured-Items-margin-top matchbox-parent">
+                    <partial name="GenericFeaturedItem" model='Model.Items.Skip(count).Take(4)' />
+                </div>
+            }
```

Comment:
if the only thing that is different here is the foreach count, instead of having an if else with the html twice, can we just set the variable for the count.

@{
var loopCount = Model.HideButton ? Model.Items.Count() : 8
}

Reply:
Done :)